### PR TITLE
SDIT-1298 Refactor OffenderResourceIntTest with DSL

### PR DESCRIPTION
**Pull request by andymarke**
Add Alias to the DSL

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt [hunk 1]
```diff
@@ -701,8 +701,7 @@ class OffenderResourceIntTest_dischargeToHospital : ResourceTest() {
 
   private fun createOffenderWithNoBooking() {
     builder.build {
-      offenderNo = offender {
-      }.offenderNo
+      offenderNo = offender().offenderNo
     }
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt:705]: minor suggestion from previous PR

File: src/test/java/uk/gov/justice/hmpps/prison/dsl/OffenderBooking.kt [hunk 2]
```diff
@@ -70,7 +70,7 @@ interface BookingDsl {
   fun transferOut(
     prisonId: String = "MDI",
     transferTime: LocalDateTime = LocalDateTime.now().minusHours(1),
-    movementReasonCode: String = "CA",
+    movementReasonCode: String = "NOTR",
     commentText: String = "Transfer",
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/dsl/OffenderBooking.kt:74]: code CA was actually invalid

**Review by lbennett-moj — COMMENTED** [review]
lgtm

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_newBooking.kt [hunk 3]
```diff
@@ -1199,16 +1199,29 @@ class OffenderResourceIntTest_newBooking : ResourceTest() {
...
+    builder.build {
+      offenderNo = offender {
+        booking(prisonId = "MDI") {
+          release()
+        }
+      }.offenderNo
+    }
+
+    return offenderNo
+  }
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_newBooking.kt:1226]: Are there any tests here where there is no booking at all - the new booking is the first one?

Comment (andymarke) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_newBooking.kt:1226]: Yes in the NewOffender nested test

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**